Improve CUTEst benchmarks

arnavk23 opened the pull request:
## Checklist

- [ ] Appropriate tests were added
- [x] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [x] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API
  
## Additional context

Add any other context about the problem here.

arnavk23 commented:
Let's check again.

ChrisRackauckas commented:
It fails to even load. Did you test it locally?

arnavk23 commented on benchmarks/OptimizationCUTEst/Project.toml:
```diff
@@ -3,6 +3,7 @@ CUTEst = "1b53aba6-35b6-5f92-a507-53c67d53f819"
 DataFrames = "a93c6f00-e57d-5684-b7b6-d8193f3e46c0"
 Ipopt = "b6b21f68-93f8-5de0-b562-5493be1d77c9"
 NLPModels = "a4795742-8479-5a88-8948-cc11e1c8c1a6"
+OMJulia = "0f4fe800-344e-11e9-2949-fb537ad918e1"
```
Okay. So what do you think of the error on Vaibhav's pr.

>ERROR: LoadError: Failed to precompile SciMLBenchmarks [31c91b34-3c75-11e9-0341-95557aab0344] to "/cache/julia-buildkite-plugin/depots/5b300254-1738-4989-ae0a-f4d2d937f953/compiled/v1.9/SciMLBenchmarks/jl_Xe9X2k".

ChrisRackauckas replied:
what commit?

ChrisRackauckas replied:
That commit doesn't add openmodelica at all.

arnavk23 replied:
That has what I have been saying chris but the error points out to openmodelica.

ChrisRackauckas replied:
This is the latest error https://buildkite.com/julialang/scimlbenchmarks-dot-jl/builds/3412#0197d4d8-ee5a-4269-ab8f-1fae086e1c49 and it does not point to openmodelica. Nor is openmodelica installed at all. Nor should it be: these benchmarks are not of DAEs, there is no modelica here.

arnavk23 replied:
Sure, will look into it and try again.

ChrisRackauckas commented:
review of what?

arnavk23 commented:
Review of the changes in Project.toml and whether they work well on the system.

arnavk23 commented:
@ChrisRackauckas any thoughts on CUTEst_safe_solvers.jmd and the changes made in this pr?

ChrisRackauckas commented:
What was changed? What do you get locally?

arnavk23 commented:
Added CUTEst_safe_solvers.jmd and corrected existing files in CUTEst by replacing deprecated .select by .select_sif_problems.

Based on my work session, this

1. demonstrated that the benchmarking infrastructure can loop through multiple optimizers (LBFGS, Ipopt) on CUTEst problems, proving the framework works for expanded solver testing.
2. confirmed that the updated CUTEst.select_sif_problems() API works correctly, accessing all 293 unconstrained problems, and that both optimizers execute without errors.

ChrisRackauckas commented:
Share the plots and tables

ChrisRackauckas commented on src/SciMLBenchmarks.jl:
```diff
@@ -2,105 +2,107 @@ module SciMLBenchmarks
-      weave(target,doctype="md2pdf",out_path=dir,args=args)
-    catch ex
-      @warn "PDF generation failed" exception=(ex, catch_backtrace())
+function weave_file(folder, file, build_list = (:script, :github))
```
any change to the base src repository should be done separately.

ChrisRackauckas commented on docs/make.jl:
```diff
@@ -8,18 +8,20 @@ dir = @__DIR__() * "/.."
-        assets=["assets/favicon.ico"],
-        canonical="https://benchmarks.sciml.ai/stable/"),
-    pages=pages
+    sitename = "The SciML Benchmarks",
```
any changes to the docs should be separate from benchmark changes.

ChrisRackauckas commented:
> Exited with status -1 (agent lost)

arnavk23 commented:
@ChrisRackauckas let's try again.

arnavk23 commented:
@ChrisRackauckas Locally also they were doing poorly but I felt as the machine had more RAM, maybe the issue wouldn't there on the machine as the code logic is there. I have tried again, take a look.

ChrisRackauckas commented:
No this machine has like 1TB of RAM. If it works on your machine it works here. Share the generated files from weaving on your machine.

arnavk23 commented:
@ChrisRackauckas try again and please review the changes made here.

ChrisRackauckas commented:
But the artifacts don't build anything, see the artifacts.

arnavk23 commented:
@ChrisRackauckas let's try again

ChrisRackauckas commented on benchmarks/OptimizationCUTEst/CUTEst_bounded.jmd:
````diff
@@ -14,34 +14,31 @@ Optimization.jl.
 This benchmark uses the following packages:
 
-```julia
+
````
This is very clearly wrong.

arnavk23 replied:
<img width="688" height="505" alt="Screenshot from 2025-07-29 04-42-54" src="https://github.com/user-attachments/assets/18a64ca0-7cd1-4557-94bb-b117005552b4" />
[CUTEst_safe_solvers.pdf](https://github.com/user-attachments/files/21478578/CUTEst_safe_solvers.pdf)

arnavk23 replied:
<img width="585" height="371" alt="Screenshot from 2025-07-29 07-15-46" src="https://github.com/user-attachments/assets/4d22947d-1755-4d25-95dd-96c13b203ab4" />

arnavk23 commented:
@ChrisRackauckas Build was failing, so tried again, if there are any errors, will be fixing again.

arnavk23 commented:
@ChrisRackauckas Build failed, tried again.

arnavk23 commented:
@ChrisRackauckas can you please look into this, it is working fine on my pc but when I am pushing it here it shows build failed even when the weave code is the original code itself.

arnavk23 commented:
@ChrisRackauckas can you please look into this, it is working fine on my pc but when I am pushing it here it shows build failed even when the weave code is the original code itself.

ChrisRackauckas commented on Project.toml:
```diff
@@ -5,13 +5,22 @@ version = "0.1.3"
 
 [deps]
 CSV = "336ed68f-0bac-5ca0-87d4-7b16caf5d00b"
+CUTEst = "1b53aba6-35b6-5f92-a507-53c67d53f819"
```
No this Project.toml shouldn't be changed.

ChrisRackauckas replied:
```suggestion
```

ChrisRackauckas commented on benchmarks/OptimizationCUTEst/CUTEst_bounded.jmd:
````diff
@@ -12,127 +12,236 @@ optimization routines. The wrapper
-This benchmark uses the following packages:
 
-```julia
+```julia; eval = true; @setup
+using Pkg; Pkg.activate("."); Pkg.instantiate()
````
```suggestion
```julia
```

ChrisRackauckas commented on benchmarks/OptimizationCUTEst/CUTEst_quadratic.jmd:
````diff
@@ -11,98 +11,262 @@ optimization routines. The wrapper
-This benchmark uses the following packages:
-
 ```julia
+using Pkg; Pkg.activate("."); Pkg.instantiate()
````
```suggestion
```

ChrisRackauckas commented on benchmarks/OptimizationCUTEst/CUTEst_unbounded.jmd:
````diff
@@ -14,72 +14,111 @@ Optimization.jl.
+# Benchmarks
+
+
+```julia; eval = true; @setup
+using Pkg; Pkg.activate("."); Pkg.instantiate()
````
```suggestion
```julia
```

ChrisRackauckas commented on benchmarks/OptimizationCUTEst/CUTEst_unconstrained.jmd:
````diff
@@ -5,96 +5,165 @@ author: Alonso M. Cisneros
+        title = "Time to solution by optimizer and number of vars",
+    )
+
+```julia; eval = true
+using Pkg; Pkg.activate("."); Pkg.instantiate()
````
```suggestion
```julia
```

ChrisRackauckas commented on benchmarks/OptimizationCUTEst/CUTEst_safe_solvers.jmd:
```suggestion
using Statistics
using StatsPlots
using StatsBase: countmap
using Printf
```

ChrisRackauckas commented on benchmarks/OptimizationCUTEst/Project.toml:
```diff
@@ -2,12 +2,17 @@
+Printf = "de0858da-6303-5e67-8744-51eddeeeb8d7"
 SciMLBenchmarks = "31c91b34-3c75-11e9-0341-95557aab0344"
 Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
+StatsBase = "2913bbd2-ae8a-5f71-8c99-4fb6c76f3a91"
```
```suggestion
Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
StatsBase = "2913bbd2-ae8a-5f71-8c99-4fb6c76f3a91"
```

ChrisRackauckas commented on benchmarks/OptimizationCUTEst/Project.toml:
```diff
@@ -2,12 +2,18 @@
+Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
+StatsBase = "2913bbd2-ae8a-5f71-8c99-4fb6c76f3a91"
 StatsPlots = "f3b207a7-027a-5e70-b257-86293d7955fd"
+Weave = "44d3d7a6-8a23-5bf8-98c5-b353f8df5ec9"
```
```suggestion
```

ChrisRackauckas commented on Project.toml:
```diff
@@ -5,13 +5,22 @@ version = "0.1.3"
+Ipopt = "b6b21f68-93f8-5de0-b562-5493be1d77c9"
 Markdown = "d6f4376e-aef5-505a-96c1-9c027394607a"
 OMJulia = "0f4fe800-344e-11e9-2949-fb537ad918e1"
+Optimization = "7f7a1694-90dd-40f0-9382-eb1efda571ba"
+OptimizationMOI = "fd9f6733-72f4-499f-8506-86b2bdd0dea1"
+OptimizationNLPModels = "064b21be-54cf-11ef-1646-cdfee32b588f"
+OptimizationOptimJL = "36348300-93cb-4f02-beb5-3c3902f8871e"
```
```suggestion
```

ChrisRackauckas commented on Project.toml:
```diff
@@ -5,13 +5,22 @@ version = "0.1.3"
+OptimizationNLPModels = "064b21be-54cf-11ef-1646-cdfee32b588f"
+OptimizationOptimJL = "36348300-93cb-4f02-beb5-3c3902f8871e"
 Pkg = "44cfe95a-1eb2-52ea-b672-e2afdf69b78f"
+Plots = "91a5bcdd-55d7-5caf-9e0b-520d859cae80"
+StatsBase = "2913bbd2-ae8a-5f71-8c99-4fb6c76f3a91"
+StatsPlots = "f3b207a7-027a-5e70-b257-86293d7955fd"
```
```suggestion
```

ChrisRackauckas commented:
oh this is a PR to a branch not to master?
